vllm/prefix.py

```diff
@@ -0,0 +1,92 @@
+from typing import List, Optional
+
+
+class Prefix:
+    """Data and states associated with a prefix of prompt tokens for multiple sequence groups.
+
+    Args:
+        prefix_id: The id of the prefix in the prefix pool.
+        token_ids: The token ids of the prefix.
+        block_size: The block size of the executed model.
+    
+    Attributes:
+        on_gpu: True if the prefix will be on GPU before the execution of the model.
+        on_cpu: True if the prefix is on CPU.
+        swap_to_gpu: True when the prefix will be computed during the execution of the model.
+    """
+
+    def __init__(
+        self,
+        prefix_id: int,
+        token_ids: List[int],
+        block_size: int,
+    ) -> None:
+        self.prefix_id = prefix_id
+        self.token_ids = token_ids
+        self.length = len(token_ids)
+        assert self.length % block_size == 0
+        self.on_gpu = False
+        self.on_cpu = False
+        self.block_table = None
```

Comment:
```suggestion
        self.block_table: Optional[List[int]] = None
```

Reply:
BTW, when will this field being set?

Reply:
It is set the block_manager.py: allocate().